transport/stunnel/server.go

```diff
@@ -0,0 +1,292 @@
+	e endpoint.Endpoint,
+	options *transport.Options) (transport.Transport, error) {
+	transportLogger := logger.WithValues("transportServer", namespacedName)
+	transferPort := e.BackendPort()
+
+	s := &server{
+		namespacedName: namespacedName,
+		options:        options,
+		listenPort:     transferPort,
+		connectPort:    stunnelConnectPort,
+		logger:         transportLogger,
+	}
+
+	err := s.reconcileConfig(ctx, c)
+	if err != nil {
+		s.logger.Error(err, "unable to reconcile stunnel server config")
+		return nil, err
+	}
+
+	err = s.reconcileSecret(ctx, c)
+	if err != nil {
+		s.logger.Error(err, "unable to reconcile stunnel server secret")
+		return nil, err
+	}
+
+	s.volumes = s.serverVolumes()
+	s.containers = s.serverContainers()
+
+	return s, nil
+}
+
+func (s *server) NamespacedName() types.NamespacedName {
+	return s.namespacedName
+}
+
+func (s *server) ListenPort() int32 {
+	return s.listenPort
+}
+
+func (s *server) ConnectPort() int32 {
+	return s.connectPort
+}
+
+func (s *server) Containers() []corev1.Container {
+	return s.containers
+}
+
+func (s *server) Volumes() []corev1.Volume {
+	return s.volumes
+}
+
+func (s *server) Type() transport.Type {
+	return TransportTypeStunnel
+}
+
+func (s *server) Credentials() types.NamespacedName {
+	return types.NamespacedName{Name: s.prefixedName(stunnelSecret), Namespace: s.NamespacedName().Namespace}
+}
+
+func (s *server) Hostname() string {
+	return "localhost"
+}
+
+func (s *server) MarkForCleanup(ctx context.Context, c ctrlclient.Client, key, value string) error {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.prefixedName(stunnelConfig),
+			Namespace: s.NamespacedName().Namespace,
+		},
+	}
+	err := utils.UpdateWithLabel(ctx, c, cm, key, value)
+	if err != nil {
+		return err
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.prefixedName(stunnelSecret),
+			Namespace: s.NamespacedName().Namespace,
+		},
+	}
+	return utils.UpdateWithLabel(context.TODO(), c, secret, key, value)
+}
+
+func (s *server) reconcileConfig(ctx context.Context, c ctrlclient.Client) error {
+	stunnelConfTemplate, err := template.New("config").Parse(stunnelServerConfTemplate)
+	if err != nil {
+		s.logger.Error(err, "unable to parse stunnel server config template")
+		return err
+	}
+
+	ports := map[string]string{
+		// acceptPort on which Stunnel service listens on, must connect with endpoint
+		"acceptPort": strconv.Itoa(int(s.ListenPort())),
+		// connectPort in the container on which Transfer is listening on
+		"connectPort": strconv.Itoa(int(s.ConnectPort())),
+	}
+	var stunnelConf bytes.Buffer
+	err = stunnelConfTemplate.Execute(&stunnelConf, ports)
+	if err != nil {
+		s.logger.Error(err, "unable to execute stunnel server config template")
+		return err
+	}
+
+	stunnelConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: s.NamespacedName().Namespace,
+			Name:      s.prefixedName(stunnelConfig),
```

Comment:
just flipping the lines to put Name first:
```suggestion
			Name:      s.prefixedName(stunnelConfig),
			Namespace: s.NamespacedName().Namespace,
```